### PR TITLE
QA-14003: Handle the display of specific format of the date picker

### PR DESCRIPTION
**Pull request by jahia-carp**

### CODE REVIEW

File: src/javascript/ContentEditor.context.js [hunk 1]
```diff
@@ -98,7 +101,8 @@ export const withContentEditorDataContextProvider = (formQuery, formDataAdapter)
             formQueryParams,
             nodeTypeName,
             nodeTypeDisplayName,
-            refetchFormData
+            refetchFormData,
+            navigatorLocale
```
Comment (jkevan) [inline src/javascript/ContentEditor.context.js:105]: I would have name it something like browserLang, and put this variable next to lang and uiLang for consistency.

File: src/javascript/SelectorTypes/DateTimePicker/DateTimePicker.jsx [hunk 2]
```diff
@@ -16,13 +16,49 @@ export const DateTimePicker = ({id, field, value, editorContext, onChange}) => {
...
+        case (dateFormat.indexOf('-') > -1):
+            separator = '-';
+            break;
+        default:
+            separator = '/';
+            break;
+    }
+
+    let regex = new RegExp('[^' + separator + ']+?', 'g');
+    let maskLocale = String(dateFormat).replace(regex, '_');
```
Comment (jkevan) [inline src/javascript/SelectorTypes/DateTimePicker/DateTimePicker.jsx:55]: Would it be easier to just replace all alphanumerical characters directly instead of trying to guess the separator ?

File: src/javascript/SelectorTypes/DateTimePicker/DateTimePicker.jsx [hunk 3]
```diff
@@ -49,7 +85,8 @@ DateTimePicker.propTypes = {
     id: PropTypes.string.isRequired,
     editorContext: PropTypes.shape({
         lng: PropTypes.string,
```
Comment (jkevan) [inline src/javascript/SelectorTypes/DateTimePicker/DateTimePicker.jsx:87]: Not part of this commit, but this look like a typo, could be great to fix it or remove the prop if it's not used.

File: src/javascript/SelectorTypes/DateTimePicker/DateTimePicker.jsx [hunk 4]
```diff
@@ -16,13 +16,49 @@ export const DateTimePicker = ({id, field, value, editorContext, onChange}) => {
...
+        'de-DE': 'DD.MM.YYYY',
+        'de-LI': 'DD.MM.YYYY',
+        'de-LU': 'DD.MM.YYYY',
+        'en-PH': 'MM/DD/YYYY',
+        'en-US': 'MM/DD/YYYY',
+        'en-ZW': 'MM/DD/YYYY',
+        'es-PA': 'MM/DD/YYYY',
+        'es-US': 'MM/DD/YYYY',
+        'zh-CN': 'YYYY/MM/DD'
+    };
```
Comment (jkevan) [inline src/javascript/SelectorTypes/DateTimePicker/DateTimePicker.jsx:35]: I would put that in a separate file to ease the update the list in case it's required in future.

File: src/javascript/SelectorTypes/DateTimePicker/DateTimePicker.jsx [hunk 5]
```diff
@@ -16,13 +16,49 @@ export const DateTimePicker = ({id, field, value, editorContext, onChange}) => {
...
+    }
+
+    let regex = new RegExp('[^' + separator + ']+?', 'g');
+    let maskLocale = String(dateFormat).replace(regex, '_');
+
+    const displayDateMask = isDateTime ? maskLocale + ' __:__' : maskLocale;
 
     return (
         <DatePickerInput
+            navigatorLocale={editorContext.navigatorLocale}
```
Comment (jkevan) [inline src/javascript/SelectorTypes/DateTimePicker/DateTimePicker.jsx:61]: Why do we pass this prop here ? I dont see any modification in DatePickerInput in current PR.
May be forgot to remove ?

File: src/javascript/SelectorTypes/DateTimePicker/DateTimePicker.jsx [hunk 6]
```diff
@@ -16,13 +16,49 @@ export const DateTimePicker = ({id, field, value, editorContext, onChange}) => {
...
+    const userNavigatorLocale = editorContext.navigatorLocale;
+
+    // Handle the date format, some languages have a specific format that we can manage here.
+    // Please note that a single digit format is not supported by our DatePicker, e.g. 'D/M/YYYY' or 'D.M.YYYY'.
+    const specificDateFormat = {
+        'de-AT': 'DD.MM.YYYY',
+        'de-CH': 'DD.MM.YYYY',
+        'de-DE': 'DD.MM.YYYY',
+        'de-LI': 'DD.MM.YYYY',
+        'de-LU': 'DD.MM.YYYY',
```
Comment (jkevan) [inline src/javascript/SelectorTypes/DateTimePicker/DateTimePicker.jsx:28]: missing: 
'de-BE'
'de-IT'

**Review by jkevan — CHANGES_REQUESTED** [review]

File: src/javascript/ContentEditor.context.js (hunk 1 above)
Comment (jahia-carp) [reply src/javascript/ContentEditor.context.js]: Done.

File: src/javascript/SelectorTypes/DateTimePicker/DateTimePicker.jsx (hunk 2 above)
Comment (jahia-carp) [reply src/javascript/SelectorTypes/DateTimePicker/DateTimePicker.jsx]: It's all me, why make it simple when you can make it complicated ;-)

File: src/javascript/SelectorTypes/DateTimePicker/DateTimePicker.jsx (hunk 3 above)
Comment (jahia-carp) [reply src/javascript/SelectorTypes/DateTimePicker/DateTimePicker.jsx]: Done.

File: src/javascript/SelectorTypes/DateTimePicker/DateTimePicker.jsx (hunk 4 above)
Comment (jahia-carp) [reply src/javascript/SelectorTypes/DateTimePicker/DateTimePicker.jsx]: Done.

File: src/javascript/SelectorTypes/DateTimePicker/DateTimePicker.jsx (hunk 5 above)
Comment (jahia-carp) [reply src/javascript/SelectorTypes/DateTimePicker/DateTimePicker.jsx]: Forgot to remove, fixed.

File: src/javascript/SelectorTypes/DateTimePicker/DateTimePicker.jsx (hunk 6 above)
Comment (jahia-carp) [reply src/javascript/SelectorTypes/DateTimePicker/DateTimePicker.jsx]: Done.

**Review by jkevan — APPROVED** [review]
